test_common/harness/deviceInfo.h

```diff
@@ -26,12 +27,28 @@ std::string get_device_info_string(cl_device_id device,
                                    cl_device_info param_name);
 
 /* Determines if an extension is supported by a device. */
-int is_extension_available(cl_device_id device, const char *extensionName);
+bool is_extension_available(cl_device_id device, const char *extensionName);
 
-/* Returns the version of the extension the device supports or throws an
- * exception if the extension is not supported by the device. */
-cl_version get_extension_version(cl_device_id device,
-                                 const char *extensionName);
+/**
+ * @brief Check if an extension is supported by an OpenCL device.
+ *
+ * In addition to checking that the extension is in the list of extensions
+ * supported by the device, the function will check the extension version.
+ * To guarantee compatibility with any breaking changes, the function
+ * succeeds only on an exact version match.
```

Comment:
I think I understand the rationale for checking for an exact version match, but I also think this is likely to cause confusion and problems in the future.  Couple of options to consider:

* Do nothing. :smile:
* Rename the function so it's more obvious that it is checking for the exact version, and leave the door open for additional overloads in the future - `is_exact_extension_version_available`?
* Add more version checking smarts, maybe something like checking for an exact version for major versions less than 1, or a matching major version and a minor version equal to or greater?

Note, we may want to ignore mismatching "patch" versions regardless.

Reply:
I don't have anything to add, but this thread made me think of this CTS Issue we still have  https://github.com/KhronosGroup/OpenCL-CTS/issues/2152 open about how the versions of provisional extensions is tested. It would be nice if we could resolve that issue as part of this PR.